src/server/rdb_save.cc

```diff
@@ -907,40 +950,86 @@ error_code RdbSaver::Impl::SaveAuxFieldStrStr(string_view key, string_view val)
   return error_code{};
 }
 
+bool RdbSaver::Impl::RecordsPopper::TryPopFromQueue() {
+  if (enforce_order && !q_records.empty() && q_records.top().id == next_record_id) {
+    record_holder = std::move(const_cast<SliceSnapshot::DbRecord&>(q_records.top()));
```

Comment:
So it indeed seems there is no way to move out of the queue and this code "should be" working as long as the pop happens immediately after.

Either we keep this, or we use the `std::make_heap` functions to maintain it ourselves. Lets ask @romange

Reply:
It seems like we have quite a few const_casts in out code base, so maybe adding this one won't hurt, but I'm generally not a fan of them

Reply:
Given that the whole case is rarely triggered during the full sync, I think simplicity and readability here are more important.  I leave Adi to decide what to do here.